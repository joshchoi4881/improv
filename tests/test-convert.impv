/*
func note[] convertIntToNote(int[] a, int n){
  int i;
  note[] notes;
  note tmp;
  int val;

  notes = [<1, "wh">];
  while(i < n-1){
    tmp = <a[i]%5, "hf">;
    notes = append(notes, [tmp]);
    i = i+1;
  }

/*
  for(i = 0; i < 2; i = i+1){
    tmp = <a[i]%5, "hf">;
    /*
    notes = append(notes, [tmp]); 
    return notes;*/
  }*/

  printNoteArr(notes);

  return notes;
} */


func note[] convertIntToNote(int[] a, int n){
  int i;
  note[] notes;
  note tmp;
  int val;

  notes = [<1, "wh">, <1, "wh">, <1, "wh">, <1, "wh">, <1, "wh">, <1, "wh">, <1, "wh">];
  
  for(i = 0; i < n-1; i = i+1){
    tmp = <a[i]%5, "hf">;
    notes[i] = tmp;
  }

  printNoteArr(notes);

  return notes;
}



func int main() {
  int[] a;
  note[] notes;

  a = [23, 11, 39, 44, 2, 16, 52];
  convertIntToNote(a, 7);
  
  /*
  
  render(convertIntToNote(a, 7), "before.mid", 1, 160);

  bubbleSort(a, 7);

  render(convertIntToNote(a, 7), "after.mid", 1, 160);*/

  return 0;
}



/*
func int main() {
  int[] a;

  a = [23, 11, 39, 44, 2, 16, 52];
  bubbleSort(a, 7);

  a = [23, 11, 39, 44, 2, 16, 52];
  selectionSort(a, 7);
  return 0;
}
*/