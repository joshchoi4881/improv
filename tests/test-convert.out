[<3, hf>, <1, hf>, <4, hf>, <4, hf>, <2, hf>, <1, hf>, <1, wh>]
